src/transformers/pipelines/audio_utils.py

```diff
@@ -167,9 +181,22 @@ def ffmpeg_microphone_live(
     else:
         chunk_s = chunk_length_s
 
-    microphone = ffmpeg_microphone(
-        sampling_rate, chunk_s, format_for_conversion=format_for_conversion, ffmpeg_input_device=ffmpeg_input_device
-    )
+    if run_in_background:
+        microphone = ffmpeg_microphone(
+            sampling_rate,
+            chunk_s,
+            format_for_conversion=format_for_conversion,
+            ffmpeg_input_device=ffmpeg_input_device,
+            background_arg="-nostdin",
+        )
+    else:
+        microphone = ffmpeg_microphone(
+            sampling_rate,
+            chunk_s,
+            format_for_conversion=format_for_conversion,
+            ffmpeg_input_device=ffmpeg_input_device,
+        )
```

Comment:
no need for big if/else here 

```suggestion
    microphone = ffmpeg_microphone(
        sampling_rate,
        chunk_s,
        format_for_conversion=format_for_conversion,
        ffmpeg_input_device=ffmpeg_input_device,
        background_arg="-nostdin" if not run_in_background else "",
    )
```